hazelcast/src/main/java/com/hazelcast/config/YamlMemberDomConfigProcessor.java

```diff
@@ -0,0 +1,866 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.config.yaml.ElementAdapter;
+import com.hazelcast.internal.yaml.YamlMapping;
+import com.hazelcast.internal.yaml.YamlNode;
+import com.hazelcast.internal.yaml.YamlScalar;
+import com.hazelcast.internal.yaml.YamlSequence;
+import com.hazelcast.util.function.Function;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.nio.ByteOrder;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.hazelcast.config.DomConfigHelper.childElements;
+import static com.hazelcast.config.DomConfigHelper.cleanNodeName;
+import static com.hazelcast.config.DomConfigHelper.getBooleanValue;
+import static com.hazelcast.config.DomConfigHelper.getIntegerValue;
+import static com.hazelcast.internal.yaml.YamlUtil.asMapping;
+import static com.hazelcast.internal.yaml.YamlUtil.asScalar;
+import static com.hazelcast.internal.yaml.YamlUtil.asSequence;
+import static com.hazelcast.util.StringUtil.lowerCaseInternal;
+import static com.hazelcast.util.StringUtil.upperCaseInternal;
+import static java.lang.Integer.parseInt;
+
+class YamlMemberDomConfigProcessor extends MemberDomConfigProcessor {
```

Comment:
As far as I understand, the parsing logic is not fully shared with `MemberDomConfigProcessor` because we have a more concise naming and structure in YAML configs, right?

Reply:
Yes, the parsing logic derives to some degree. The reason is the differences between XML and YAML. The main goal was to share as much as possible between the XML and YAML parsing logic, but without sacrificing the advantages of YAML. So instead of 
```
hazelcast:
  - map: 
    name: mapName
  - map: 
    name: mapName2
    ...
```
, the following has chosen:
```
hazelcast:
  map:
    mapName: 
       ...
    mapName2:
       ...
```
There are other parts need overriden parsing logic in the YAML case, but this is the most typical reason.